controller/konnect/index_credentials_apikey.go

```diff
@@ -0,0 +1,32 @@
+package konnect
+
+import (
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	configurationv1alpha1 "github.com/kong/kubernetes-configuration/api/configuration/v1alpha1"
+)
+
+const (
+	// IndexFieldKongCredentialAPIKeyReferencesKongConsumer is the index name for KongCredentialAPIKey -> Consumer.
+	IndexFieldKongCredentialAPIKeyReferencesKongConsumer = "kongCredentialsAPIKeyConsumerRef"
+)
+
+// IndexOptionsForCredentialsAPIKey returns required Index options for KongCredentialAPIKey.
+func IndexOptionsForCredentialsAPIKey() []ReconciliationIndexOption {
+	return []ReconciliationIndexOption{
+		{
+			IndexObject:  &configurationv1alpha1.KongCredentialAPIKey{},
+			IndexField:   IndexFieldKongCredentialAPIKeyReferencesKongConsumer,
+			ExtractValue: kongKongCredentialAPIKeyReferencesConsumer,
+		},
+	}
+}
+
+// kongKongCredentialAPIKeyReferencesConsumer returns the name of referenced Consumer.
+func kongKongCredentialAPIKeyReferencesConsumer(obj client.Object) []string {
```

Comment:
Would the name `kongKongCredentialAPIKeyReferencesConsumer` be redundant?

Reply:
What do you mean by redundant here? 🤔 Can you elaborate?

Reply:
I mean does the name `KongCredentialAPIKeyReferencesConsumer` look better? two consecutive `kong`s looks redundant.

Reply:
Ah this 😅 I haven't noticed this for some reason. 5285f498d6439257e71b6833b0f6de04569cc3b7 fixes that.